src/snowflake/cli/plugins/nativeapp/codegen/sandbox.py

```diff
@@ -0,0 +1,201 @@
+    timeout: Optional[int] = None,
+) -> subprocess.CompletedProcess:
+    conda_env = env_name
+    if conda_env is None:
+        conda_env = _get_active_conda_env()
+    if conda_env is None:
+        raise SandboxExecutionError("No conda environment found")
+
+    conda_exec = shutil.which("conda")
+    if conda_exec is None:
+        raise SandboxExecutionError(
+            "conda command not found, make sure it is installed on your system and in your PATH"
+        )
+
+    # conda run removes the need to activate the environment, as would typically be done in an interactive shell
+    return _execute_python_interpreter(
+        [conda_exec, "run", "-n", conda_env, "--no-capture-output", "python"],
+        script_source,
+        timeout=timeout,
+        cwd=cwd,
+    )
+
+
+def _execute_with_system_path_python(
+    script_source: str,
+    cwd: Optional[Union[str, Path]] = None,
+    timeout: Optional[int] = None,
+) -> subprocess.CompletedProcess:
+    python_executable = (
+        shutil.which("python3") or shutil.which("python") or sys.executable
+    )
+
+    return _execute_python_interpreter(
+        python_executable, script_source, timeout=timeout, cwd=cwd
+    )
+
+
+class ExecutionEnvironmentType(Enum):
+    AUTO_DETECT = 0  # auto-detect the current python interpreter by looking for an active virtual environment
+    VENV = 1  # use the python interpreter specified by a venv environment
+    CONDA = 2  # use the python interpreter specified by a conda environment
+    SYSTEM_PATH = 3  # search for a python interpreter in the system path
+    CURRENT = 4  # Use the python interpreter that is currently executing (i.e. `sys.executable`)
+
+
+def execute_script_in_sandbox(
+    script_source: str,
+    env_type: ExecutionEnvironmentType = ExecutionEnvironmentType.AUTO_DETECT,
+    cwd: Optional[Union[str, Path]] = None,
+    timeout: Optional[int] = None,
+    **kwargs,
+) -> subprocess.CompletedProcess:
+    """
+    Executes a python script in a sandboxed environment, and returns its output. The script is executed in a different
+    process. The execution environment is determined by the `env_type` argument. By default, the logic will attempt
+    to auto-detect the correct environment by looking for an active venv or conda environment. If none can be found, it
+    will use the system's default python executable, as determined by the user's path. As a last resort, the current
+    python execution environment will be used (still in a subprocess).
+
+    Parameters:
+        script_source (str): The python script to be executed, as a string.
+        env_type: The type of execution environment to use (default: ExecutionEnvironmentType.AUTO_DETECT).
+        cwd (Optional[Union[str, Path]]): An optional path to use as the current directory when exeecuting the script.
+        timeout (Optional[int]): An optional timeout in seconds when executing the script. Defaults to no timeout.
```

Comment:
For both `timeout` and `cwd`, are these values ever going to differ from their defaults? If yes, where can a user provide them, the PDF file?

Reply:
Cam got what I was trying to do. This is a general function, so it's not too opinionated about these params.

Reply:
Ack